pkg/volume/csi/csi_mounter.go

```diff
@@ -375,21 +374,24 @@ func isDirMounted(plug *csiPlugin, dir string) (bool, error) {
 // removeMountDir cleans the mount dir when dir is not mounted and removed the volume data file in dir
 func removeMountDir(plug *csiPlugin, mountPath string) error {
 	klog.V(4).Info(log("removing mount path [%s]", mountPath))
+	notMount := bool(false)
 	if pathExists, pathErr := util.PathExists(mountPath); pathErr != nil {
 		klog.Error(log("failed while checking mount path stat [%s]", pathErr))
 		return pathErr
 	} else if !pathExists {
-		klog.Warning(log("skipping mount dir removal, path does not exist [%v]", mountPath))
-		return nil
+		klog.Warning(log("mount dir removal, path does not exist [%v]", mountPath))
+		notMount = true
```

Comment:
For example: 
Here is the mount info
```
/dev/mapper/vg10000-vg10000_pvc--dfdb08b5f38011e8 on /var/mnt/kubelet/pods/ee16af32-f380-11e8-b5f3-74dbd180905c/volumes/kubernetes.io~csi/pvc-dfdb08b5f38011e8/mount
```
There are `vol_data.json` file which need to be cleaned as well under dir  ` /var/mnt/kubelet/pods/ee16af32-f380-11e8-b5f3-74dbd180905c/volumes/kubernetes.io~csi/pvc-dfdb08b5f38011e8/` 
The dir  mountPath will be  `/var/mnt/kubelet/pods/ee16af32-f380-11e8-b5f3-74dbd180905c/volumes/kubernetes.io~csi/pvc-dfdb08b5f38011e8/mount` when do tearDown. If the `mount` dir had been deleted, but `vol_data.json` file still exists under dir ``/var/mnt/kubelet/pods/ee16af32-f380-11e8-b5f3-74dbd180905c/volumes/kubernetes.io~csi/pvc-dfdb08b5f38011e8/` , then it will not be removed, and dir `/var/mnt/kubelet/pods/ee16af32-f380-11e8-b5f3-74dbd180905c/volumes/kubernetes.io~csi/pvc-dfdb08b5f38011e8/` also can't be removed if we just return here.

Reply:
Oh, I see, there is `volPath := path.Dir(mountPath)` and its cleanup below, hidden in the diff. Sorry.